ui/app/controllers/settings/tokens.js

```diff
@@ -90,13 +90,20 @@ export default class Tokens extends Controller {
     window.localStorage.setItem('nomadOIDCNonce', nonce);
     window.localStorage.setItem('nomadOIDCAuthMethod', provider);
 
+    let redirectURL;
+    if (Ember.testing) {
+      redirectURL = this.router.currentURL;
+    } else {
+      redirectURL = new URL(window.location.toString());
+      redirectURL.search = '';
+      redirectURL = redirectURL.href;
```

Comment:
`nit`: This line nullifies the work you've just done on line 97 and 98, no?

Reply:
What's hard for me to grok here is that, I'm reading this as:

```
variable `redirectURL` is assigned to a new object in memory using URL constructor
`redirectURL.search` is assigned a value of empty string
`redirectURL` is assigned to the value of `redirectURL.href` -- this would override what just happened in the previous step, no?
```

Reply:
If the URL constructor generated a POJO or something that would be true, but it differs in a few ways.

Consider if I was running this from a hypothetical `site.com/page?foo=bar`

```
redirectURL = { href:"site.com/page?foo=bar", search="foo=bar", ... }
redirectURL.search = "" // since URL.href is dependent upon search and other props, it is modified to become "site.com/page"
redirectURL = "site.com/page"
```

Reply:
Aha! Couldn't we have just take `redirectURL.host` or `redirectURL.origin` instead? But either way, that's clear now.